src/Elliptic/Systems/Elasticity/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,128 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class Elasticity::FirstOrderSystem
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "Elliptic/Systems/Elasticity/Equations.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "ParallelAlgorithms/LinearSolver/Tags.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/ConstitutiveRelation.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Tags {
+template <typename>
+class Variables;
+}  // namespace Tags
+
+namespace LinearSolver {
+namespace Tags {
+template <typename>
+struct Operand;
+}  // namespace Tags
+}  // namespace LinearSolver
+/// \endcond
+
+namespace Elasticity {
+
+/*!
+ * \brief The Elasticity equation formulated as a set of coupled first-order
+ * PDEs.
+ *
+ * \details This system formulates the Elasticity equation \f$\nabla T(x) =
+ * f(x)\f$ on a euclidean space as the set of coupled first-order PDEs
+ *
+ * \f[
+ * \nabla \boldsymbol{T}(x) = \boldsymbol{f}(x) \\
+ * -\frac{\nabla \boldsymbol{\xi}(x) + (\nabla \boldsymbol{\xi}(x))^T}{2} +
+ * \boldsymbol{S}(x) = 0 \f]
+ *
+ * with the stress tensor T given by \f$T^{ij}(x) = -Y^{ijkl}S_{kl}(x)\f$,
+ * \f$Y^{ijkl}\f$ being the material-specific Youngs tensor and the strain given
+ * by \f$S_{kl}(x) = \partial_{(k}\xi_{l)}(x)\f$ as an auxiliary variable.
+ *
+ *
+ * This scheme also goes by the name of _mixed_ or _flux_ formulation (see e.g.
+ * \cite Arnold2002). The reason for the latter name is that we can write the
+ * set of coupled first-order PDEs in flux-form
+ *
+ * for the primary equation
+ *
+ * \f[
+ * \partial_i F^i_A + S_A = f_A(x)
+ * * \f]
+ *
+ *  and respectively, for the auxiliary equation
+ *
+ * \f[
+ * -\partial_i F^i_A + S_A = f_A(x)
+ * \f]
+ *
+ * by choosing the fluxes and sources in terms of the system variables
+ * \f$\xi_j(x)\f$ and \f$S_{kl}(x)\f$ as
+ *
+ * \f{align*}
+ * F^i_{\xi_j} &=  -Y^{ijkl}_{(\Chi, \\xi, S)} S_{kl}(x) \\
```

Comment:
I have written xi with upper indices where it stands on its own. To avoid confusion, I kept lower index in the definition of the strain. Hope that's alright.